### PR TITLE
Feature/1786 - Fix Date issues

### PR DESCRIPTION
**Pull request by sasha-dresden**
Issue [FECFILE-1786](https://fecgov.atlassian.net/browse/FECFILE-1786)

### CODE REVIEW

File: front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.ts [hunk 1]
```diff
@@ -136,9 +134,9 @@ export class CreateF3XStep1Component extends DestroyerComponent implements OnIni
           isElectionYear,
           filingFrequency,
         );
-        this.form.patchValue({ coverage_from_date, coverage_through_date });
-      } else {
-        this.form.patchValue({ coverage_from_date: null, coverage_through_date: null });
+
+        this.coverageFromDate.control.setValue(coverage_from_date);
```
Comment (toddlees) [inline front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.ts]: Does this new approach not let us patchValues the way we did before?

File: front-end/src/app/shared/components/inputs/amount-input/amount-input.component.html [hunk 2]
```diff
@@ -7,42 +7,20 @@
...
-          <p-calendar
-            (onFocus)="calendarOpened = true"
-            (onClose)="validateDate(templateMap['date'], false)"
-            (onBlur)="validateDate(templateMap['date'], calendarOpened)"
-            inputId="date"
-            name="date"
-            [formControlName]="templateMap['date']"
-          ></p-calendar>
-          <app-error-messages
-            *ngIf="!templateMap['date2']"
```
Comment (toddlees) [inline front-end/src/app/shared/components/inputs/amount-input/amount-input.component.html:40]: Looks like we need to not show an error if there's no date2

File: front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.html [hunk 3]
```diff
@@ -130,40 +124,22 @@ <h3 class="m-0">Covering period</h3>
...
-            name="coverage_from_date"
-            formControlName="coverage_from_date"
-          ></p-calendar>
-          <app-error-messages
-            [form]="form"
-            fieldName="coverage_from_date"
-            [formSubmitted]="formSubmitted"
-          ></app-error-messages>
-        </div>
+        <app-calendar
```
Comment (toddlees) [inline front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.html:127]: validating the to date on from date change doesn't seem to be working anymore
image
On DEV:
image

File: front-end/src/app/shared/components/inputs/loan-terms-dates-input/loan-terms-dates-input.component.ts [hunk 4]
```diff
@@ -142,24 +142,30 @@ export class LoanTermsDatesInputComponent extends BaseInputComponent implements
...
     }
   }
 
+  /**
+   * Alternates between a date form control and a string form control.
+   * The new CalendarComponent replaces the string form control with a Date control.
+   * This happens on Init, which means when the ngIf condition is met.
+   * This means when returning to the string control we need to recreate the control.
+   * @param newDueDateSetting
+   */
```
Comment (toddlees) [inline front-end/src/app/shared/components/inputs/loan-terms-dates-input/loan-terms-dates-input.component.ts:170]: This conversion should no longer happen.  Switching the type should clear the value https://fecgov.atlassian.net/browse/FECFILE-1786

**Review by toddlees — CHANGES_REQUESTED** [review]

File: front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.ts (hunk 1 above)
Comment (sasha-dresden) [reply front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.ts]: After checking patchValues still works just fine. This was a bit of a leftover from when I was first trying to figure out how to handle losing validators when a new control is created. Returned it to the old patchValues. Also returned the patch value to null else condition.

File: front-end/src/app/shared/components/inputs/amount-input/amount-input.component.html (hunk 2 above)
Comment (sasha-dresden) [reply front-end/src/app/shared/components/inputs/amount-input/amount-input.component.html:40]: Oh good catch. In order to resolve this and the custom error message for app-error-messages I needed to pass two additional fields to the calendar, but doing so resolved both issues.

File: front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.html (hunk 3 above)
Comment (sasha-dresden) [reply front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.html:127]: Oh this was tricky. I thought I'd caught all of the validator issues when I directly copied the validators from the old control to the new one, but this showed a second area of problem. When the validator in question was taking in a second control and that second control no longer exists. Was able to fix this by passing the form and the field name to allow the validator to retrieve the correct control at validation time. This is something we'll want to be aware of in the future if we ever add any additional validators comparing two date controls.

File: front-end/src/app/shared/components/inputs/loan-terms-dates-input/loan-terms-dates-input.component.ts (hunk 4 above)
Comment (sasha-dresden) [reply front-end/src/app/shared/components/inputs/loan-terms-dates-input/loan-terms-dates-input.component.ts:170]: Ah, I totally misread that. I thought it was specifically talking about only the error messages being cleared, not the values themselves as well. I made some updates so now both the due date field and the interest rate field get set to empty when the associated setting field is changed.

**Comment by toddlees** [comment]
works beautifully

**Review by toddlees — APPROVED** [review]

**Pull request merged**